tests/integration/targets/dhcp_option_code/tasks/main.yml

```
@@ -0,0 +1,253 @@
+---
+- module_defaults:
+    group/infoblox.universal_ddi.all:
+      portal_url: "{{ portal_url }}"
+      portal_key: "{{ portal_key }}"
+  block:
+    - ansible.builtin.set_fact:
+        option_code_name: "basic_opt_code_1"
+        option_code: 234
+        option_item_type: "boolean"
+
+    - name: Create a DHCP Option Code (check mode)
+      infoblox.universal_ddi.dhcp_option_code:
+        code: 234
```

```suggestion
        code: "{{ option_code }}"
```
Please change all the occurrences of option_code to use option_code variable

---

it expects int and type casting does work as expected